var/spack/repos/builtin/packages/dftbplus/package.py

```diff
@@ -3,85 +3,162 @@
         default=False,
-        description="Use the MAGMA library " "for GPU accelerated computation",
+        description="Whether the ARPACK library should be included (needed for TD-DFTB).",
+        when="~mpi",
+    )
+    variant(
+        "chimes",
+        default=False,
+        when="@21.2:",
+        description="Whether repulsive corrections" "via the ChIMES library should be enabled.",
     )
-
     variant(
         "elsi",
         default=False,
         description="Use the ELSI library for large scale systems. "
         "Only has any effect if you build with '+mpi'",
+        when="+mpi",
+    )
+    variant(
+        "gpu",
+        default=False,
+        description="Use the MAGMA library " "for GPU accelerated computation",
+    )
+    variant(
+        "mbd",
+        default=False,
+        when="@21.1:",
+        description="Whether DFTB+ should be built with many-body-dispersion support.",
+    )
+    variant("mpi", default=False, description="Whether DFTB+ should support MPI-parallelism.")
+    variant(
+        "openmp",
+        default=True,
+        description="Whether OpenMP thread parallisation should be enabled.",
+    )
+    variant(
+        "plumed",
+        default=False,
+        when="@20.1:",
+        description="Whether metadynamics via the PLUMED2 library should be allowed.",
+    )
+    variant("poisson", default=False, description="Whether the Poisson-solver should be included.")
+    variant(
+        "python",
+        default=False,
+        description="Whether the Python components of DFTB+ should be tested and installed.",
+    )
+    variant(
+        "sdftd3",
+        default=False,
+        when="@21.2:",
+        description="Whether the s-dftd3 library should be included",
     )
-
     variant(
         "sockets",
         default=False,
         description="Whether the socket library " "(external control) should be linked",
     )
-
-    variant("arpack", default=False, description="Use ARPACK for excited state DFTB functionality")
-
     variant(
         "transport",
         default=False,
+        when="+shared",
         description="Whether transport via libNEGF should be included. "
         "Only affects parallel build. "
         "(serial version is built without libNEGF/transport)",
     )
+    variant(
+        "tblite",
+        default=False,
+        when="@21.2:",
+        description="Whether xTB support should be included via tblite.",
+    )
+
+    variant("shared", default=False, description="Most often for the Python wrappers.")
 
+    # ONLY for version 19.1
     variant(
         "dftd3",
         default=False,
+        when="@:21.1",
         description="Use DftD3 dispersion library " "(if you need this dispersion model)",
     )
 
-    depends_on("lapack")
-    depends_on("blas")
-    depends_on("scalapack", when="+mpi")
-    depends_on("mpi", when="+mpi")
+    depends_on("cmake@3.16:", type="build")
+    depends_on("ninja@1.10", type="build")
```

Comment:
I think you want to use perhaps the ``when` context manager for dependencies that are specific to either build system.  There's an example in the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#conditional-dependencies.

Reply:
Ah yes, missed this
```suggestion
    depends_on("cmake", type="build", when="@20.1:")
    depends_on("ninja@1.10", type="build", when="@20.1:")
```